cmd/xl-v1-multipart-common.go

```diff
@@ -19,14 +19,13 @@ package cmd
 import (
 	"path"
 	"sync"
+	"time"
 )
 
-// writeUploadJSON - create `uploads.json` or update it with change
-// described in uCh.
-func (xl xlObjects) updateUploadJSON(bucket, object string, uCh uploadIDChange) error {
+// updateUploadJSON - add or remove upload ID info in all `uploads.json`.
+func (xl xlObjects) updateUploadJSON(bucket, object, uploadID string, initiated *time.Time, isRemove bool) error {
```

Comment:
I see `initiated` is used only if `isRemove` is `false`.  I don't mind of using a value than a pointer.  But I don't understand why pointer using is not idiomatic Go style.  Its all depending on the context.  The question here is whether we want to allocate dummy structure or not.